tests/mypy_test.py

```diff
@@ -183,9 +186,24 @@ def run_mypy(
         if args.dry_run:
             exit_code = 0
         else:
-            stdout, stderr, exit_code = mypy_run(mypy_args)
-            print(stdout, end="")
-            print(stderr, end="")
+            stdout_redirect, stderr_redirect = StringIO(), StringIO()
+            with redirect_stdout(stdout_redirect), redirect_stderr(stderr_redirect):
+                returned_stdout, returned_stderr, exit_code = mypy_run(mypy_args)
+
+            if exit_code:
+                print_error("failure\n")
+                captured_stdout = stdout_redirect.getvalue()
+                captured_stderr = stderr_redirect.getvalue()
+                if returned_stderr:
+                    print_error(returned_stderr)
+                if captured_stderr:
+                    print_error(captured_stderr)
+                if returned_stdout:
+                    print_error(returned_stdout)
+                if captured_stdout:
+                    print_error(captured_stdout, end="")
```

Comment:
Should we really color the whole error output red? This could be a bit hard to read.

Reply:
Hmm. We could just colour the `failure` message at the beginning red. But I don't find https://github.com/python/typeshed/runs/6505698999?check_suite_focus=true too hard to read, and I like how easy it is for me to see where the error messages start and where they end.